src/panels/alarm-table-react/AlarmTableControl.tsx

```diff
@@ -0,0 +1,82 @@
+import { PanelProps } from '@grafana/data';
+import { Button, ContextMenu, Modal, Pagination, Tab, TabContent, Table, TabsBar } from '@grafana/ui';
+
+import React from 'react'
+import { AlarmTableMenu } from './AlarmTableMenu';
+import { AlarmTableModalContent } from './modal/AlarmTableModalContent';
+import { AlarmTableSelectionStyles } from './AlarmTableSelectionStyles';
+import { AlarmTableControlProps } from './AlarmTableTypes';
+import { useAlarmHelmProperties } from './hooks/useAlarmHelmProperties';
+import { useAlarmTableMenuActions } from './hooks/useAlarmTableMenuActions';
+import { useAlarmTableConfigDefaults } from './hooks/useAlarmTableConfigDefaults';
+import { useAlarmTableMenu } from './hooks/useAlarmTableMenu';
+import { useAlarmTableRowHighlighter } from './hooks/useAlarmTableRowHighlighter';
+import { useAlarmTableSelection } from './hooks/useAlarmTableSelection';
+import { useAlarmTableModalTabs } from './hooks/useAlarmTableModalTabs';
+import { useOpenNMSClient } from './hooks/useOpenNMSClient';
+import { useAlarm } from './hooks/useAlarm';
+
+
+export const AlarmTableControl: React.FC<PanelProps<AlarmTableControlProps>> = (props) => {
+
+
+    const { state, rowClicked, soloIndex } = useAlarmTableSelection(() => {
+        setDetailsModal(true)
+    });
+
+    const { table, menu, menuOpen, setMenuOpen } = useAlarmTableMenu(rowClicked);
+    const { actions, detailsModal, setDetailsModal } = useAlarmTableMenuActions(state.indexes, () => setMenuOpen(false));
+    const { tabActive, tabClick, resetTabs } = useAlarmTableModalTabs();
+    const { client } = useOpenNMSClient(props.data?.request?.targets?.[0]?.datasource)
+    const { alarm, goToAlarm, alarmQuery } = useAlarm(props?.data?.series, soloIndex, client);
+    const { filteredProps, page, setPage, totalPages } = useAlarmHelmProperties(props?.data?.series[0], props?.options?.alarmTable);
+
+
+    useAlarmTableRowHighlighter(state, table);
+    useAlarmTableConfigDefaults(props.fieldConfig, props.onFieldConfigChange, props.options)
+    const getFontSize = () => {
+        const fontSize = props.options?.alarmTable.alarmTablePaging.fontSize?.value
+        return `font-size-${fontSize}`
+    }
+    return (
+        <div ref={table} className={
+            `
+            ${alarmQuery ? 'alarm-query' : 'non-alarm-query'}
+            ${props.options?.alarmTable?.alarmTablePaging?.scroll ? 'scroll' : ' no-scroll'}
```

Comment:
Remove space in `' no-scroll'` ? I assume line breaks in the template string will add a space?

Reply:
(ok as is)